### PR TITLE
feature: save search result into org file

### PR DESCRIPTION
**Pull request by kaiwk**
- make bing-dict can save search result into org file, build user's own vocabulary book.
- user also can toggle this function by set `bing-dict--save-search-result` nil.

### CODE REVIEW

File: bing-dict.el [hunk 1]
```diff
@@ -133,8 +133,52 @@ The value could be `synonym', `antonym', `both', or nil.")
                                          'face
                                          'font-lock-builtin-face))
 
+(defvar bing-dict--org-file "~/.orgs/vocabulary.org"
```
Comment (cute-jumper) [inline bing-dict.el]: The default path should be something consisting of `user-emacs-directory` and `bing-dict`. Say "~/.emacs.d/bing-dict/".

File: bing-dict.el [hunk 2]
```diff
@@ -133,8 +133,52 @@ The value could be `synonym', `antonym', `both', or nil.")
...
+  "The file where store the vocabulary.")
+
+(defvar bing-dict--org-file-title "Vocabulary"
```
Comment (cute-jumper) [inline bing-dict.el]: Users should be able to set this variable and the following variable, thus the names of these variables should not contain `--`. Use the single dash instead.

File: bing-dict.el [hunk 3]
```diff
@@ -133,8 +133,52 @@ The value could be `synonym', `antonym', `both', or nil.")
...
+
+(defvar bing-dict--save-search-result nil
+  "save bing dict search result or not.")
+
+
+(defun bing-dict--tidy-headlines ()
+  "remove extra spaces between stars and the headline text"
+  (save-excursion
+    (goto-char (point-min))
+    (while (re-search-forward "^\\*+\\([[:space:]][[:space:]]+\\)" (point-max) t)
```
Comment (cute-jumper) [inline bing-dict.el:150]: Can this regexp be replaced by `org-heading-regexp`?

File: bing-dict.el [hunk 4]
```diff
@@ -133,8 +133,52 @@ The value could be `synonym', `antonym', `both', or nil.")
...
+      (replace-match " " nil nil nil 1))))
+
+(defun bing-dict--save-word (word definition)
+  "save word in org file"
+  (find-file bing-dict--org-file)
```
Comment (cute-jumper) [inline bing-dict.el]: If you want to write to a file, it would be better to use `write-region`.

**Comment by cute-jumper** [comment]
Thanks! I've added some comments regarding your PR. Can you try to address these problems when you're free?

File: bing-dict.el (hunk 3 above)
Comment (kaiwk) [reply bing-dict.el:150]: No, it seems can't, `org-heading-regexp` matches two groups -- stars are put in group 1 and the trimmed body in group 2. But in this case,
 
>"remove extra spaces between stars and the headline text" 

should only match the "space" after the stars.

This function is aim to make the vocabularies more good-looking in org mode.

File: bing-dict.el (hunk 2 above)
Comment (kaiwk) [reply bing-dict.el]: OK, I will fix that~

File: bing-dict.el (hunk 4 above)
Comment (kaiwk) [reply bing-dict.el]: OK, I will try it.
Thank you for your work!

**Comment by cute-jumper** [comment]
Thanks! I'll merge this now and fix a few small issues later by myself. Thanks for the contribution!

**Comment by cute-jumper** [comment]
I refactored the code a little bit. If you find any problem, please feel free to discuss it. And also, the documentation needs to be updated. Thanks!

**Pull request merged**